gef.py

```diff
@@ -3679,20 +3679,24 @@ def reset_architecture(arch: Optional[str] = None) -> None:
             raise OSError(f"Specified arch {arch.upper()} is not supported")
         return
 
-    gdb_arch = get_arch()
+    # check for bin running
+    if is_alive():
+        try:
+            arch = gdb.selected_frame().architecture()
+            gef.arch = arches[arch.name()]()
+        except KeyError:
+            raise OSError(f"Unsupported Architecture {arch.name().upper()} from binary")
+        return
 
-    preciser_arch = next((a for a in arches.values() if a.supports_gdb_arch(gdb_arch)), None)
```

Comment:
I didn't originally realize you removed this, and this part is important: using `supports_gdb_arch` allow a specific to enforce an architecture. It should not be removed.

Reply:
Sorry about this, I got it back on live case.